launcher/minecraft/auth/Parsers.cpp

```diff
@@ -212,6 +212,180 @@ bool parseMinecraftProfile(QByteArray & data, MinecraftProfile &output) {
     return true;
 }
 
+namespace {
+    // these skin URLs are for the MHF_Steve and MHF_Alex accounts (made by a Mojang employee)
+    // they are needed because the session server doesn't return skin urls for default skins
+    static const QString SKIN_URL_STEVE = "http://textures.minecraft.net/texture/1a4af718455d4aab528e7a61f86fa25e6a369d1768dcb13f7df319a713eb810b";
+    static const QString SKIN_URL_ALEX = "http://textures.minecraft.net/texture/83cee5ca6afcdb171285aa00e8049c297b2dbeba0efb8ff970a5677a1b644032";
+
+    bool isDefaultModelSteve(QString uuid) {
+        // need to calculate *Java* hashCode of UUID
+        // if number is even, skin/model is steve, otherwise it is alex
+
+        // just in case dashes are in the id
+        uuid.remove('-');
+
+        if (uuid.size() != 32) {
+            return true;
+        }
+
+        // qulonglong is guaranteed to be 64 bits
+        // we need to use unsigned numbers to guarantee truncation below
+        qulonglong most = uuid.left(16).toULongLong(nullptr, 16);
+        qulonglong least = uuid.right(16).toULongLong(nullptr, 16);
+        qulonglong xored = most ^ least;
+        return ((static_cast<quint32>(xored >> 32)) ^ static_cast<quint32>(xored)) % 2 == 0;
+    }
+}
+
+/**
+Uses session server for skin/cape lookup instead of profile,
+because locked Mojang accounts cannot access profile endpoint
+(https://api.minecraftservices.com/minecraft/profile/)
+
+ref: https://wiki.vg/Mojang_API#UUID_to_Profile_and_Skin.2FCape
+
+{
+    "id": "<profile identifier>",
+    "name": "<player name>",
+    "properties": [
+        {
+            "name": "textures",
+            "value": "<base64 string>"
+        }
+    ]
+}
+
+decoded base64 "value":
+{
+    "timestamp": <java time in ms>,
+    "profileId": "<profile uuid>",
+    "profileName": "<player name>",
+    "textures": {
+        "SKIN": {
+            "url": "<player skin URL>"
+        },
+        "CAPE": {
+            "url": "<player cape URL>"
+        }
+    }
+}
+*/
+
+bool parseMinecraftProfileMojang(QByteArray & data, MinecraftProfile &output) {
+    qDebug() << "Parsing Minecraft profile...";
+#ifndef NDEBUG
+    qDebug() << data;
+#endif
+
+    QJsonParseError jsonError;
+    QJsonDocument doc = QJsonDocument::fromJson(data, &jsonError);
+    if(jsonError.error) {
+        qWarning() << "Failed to parse response as JSON: " << jsonError.errorString();
+        return false;
+    }
+
+    auto obj = doc.object();
```

Comment:
include our `Json.h` and:
```suggestion
    auto obj = Json::requireObject(doc, "mojang minecraft profile");
```